collector.go

```diff
@@ -192,6 +196,9 @@ func pduToSample(indexOids []int, pdu *gosnmp.SnmpPDU, metric *config.Metric, oi
 		t = prometheus.GaugeValue
 		value = 1.0
 		stringType = true
+		if len(metric.Overrides.StringMetrics) > 0 {
+			res = generateOverrideMetrics(metric, pdu.Value.(string))
```

Comment:
You still need all the label logic below to apply. I'd reorder the code so this flows more naturally.